src/gt4py/next/program_processors/runners/dace_iterator/__init__.py

```diff
@@ -12,9 +12,18 @@
 #
 # SPDX-License-Identifier: GPL-3.0-or-later
 
+try:
+    import dace
+except ModuleNotFoundError:
+    import sys
+
+    if "pytest" in sys.modules:
+        raise
+    print("Error, module dace not found", file=sys.stderr)
+    exit()
```

Comment:
Why is the `import dace` needed here? The importer of this package (either a test module or another part of gt4py) should decide how deal with the import exception on its own, right?

Reply:
It seems that the Spack tests are executing this command:
`python3.10 -c 'import gt4py.next.program_processors.runners.dace_iterator'`
I received the following error report from @abishekg7 for Spack build, before I applied the change in this PR:
```
==> Testing package py-gt4py-test_dace-bab6rui
==> [2023-09-12-11:21:09.665161] Running install-time tests
==> [2023-09-12-11:21:09.665311] ('RUN-TESTS: install-time tests [test]',)
==> Error: TestFailure: 1 tests failed.


Command exited with status 1:
    '/project/g110/spack/upstream/daint_v0.18.1.7/icon-dsl/python-3.10.4/gcc-9.3.0/3enasifkvryqtb57gwbaodoxbyobiloa/bin/python3.10' '-c' 'import gt4py.next.program_processors.runners.dace_iterator'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/scratch/snx3000/agopal/spack_test/test/spack/opt/spack/py-gt4py-test_dace/gcc-9.3.0/bab6ruizwdfnhttn4oyq5jwgh4vzrkhq/lib/python3.10/site-packages/gt4py/next/program_processors/runners/dace_iterator/__init__.py", line 17, in <module>
    import dace
ModuleNotFoundError: No module named 'dace'
```

Reply:
I have tried the Spack build locally with this PR and I observed that the change discussed here is not needed. The PR seems to work fine without this change, so I am going to remove it in next commit.